core/simulations/helpers/executeSpell.ts

```diff
@@ -0,0 +1,71 @@
+import BigNumber from '../../src/bignumber';
+import hre from 'hardhat';
+import { TEST_NETWORK } from '../../helpers/constants';
+import getContract from '../../src/contracts';
+import getSigner from '../../src/signer';
+import { overwriteUintMapping } from '../../helpers/hardhat/slotOverwrite';
+import { MKR_NUMBER_OF_DIGITS } from '../../src/constants/UNITS';
+
+const ethers = hre.ethers;
+const provider = hre.network.provider;
+
+const getMaximumVotingPoverInMkr = async () => {
+    const chiefContract = await getContract(TEST_NETWORK, 'MCD_ADM');
+    const hatAddress = await chiefContract.hat();
+    const approvals = await chiefContract.approvals(hatAddress);
+    return new BigNumber(approvals._hex).shiftedBy(-MKR_NUMBER_OF_DIGITS);
+};
+
+const setGovernanceMkr = async (mkrAmount: BigNumber) => {
+    const signer = await getSigner(TEST_NETWORK);
+    const currentAddress = await signer.getAddress();
+    const govContract = await getContract(TEST_NETWORK, 'MCD_GOV');
+    const currentBalance = await govContract.balanceOf(currentAddress);
+    await overwriteUintMapping('MCD_GOV', '0x01', currentAddress, mkrAmount.shiftedBy(MKR_NUMBER_OF_DIGITS));
+    const newBalance = await govContract.balanceOf(currentAddress);
+    if (newBalance._hex === currentBalance._hex) {
+        throw new Error(
+            `governance MKR balance doesn't seem to change: ${ethers.utils.formatEther(
+                currentBalance
+            )}, ${ethers.utils.formatEther(newBalance)}`
+        );
+    }
+};
+
+const voteForSpell = async (spellAddress: string, mkrAmount: BigNumber) => {
+    const govContract = await getContract(TEST_NETWORK, 'MCD_GOV', true);
+    const chiefContract = await getContract(TEST_NETWORK, 'MCD_ADM', true);
+    const mkrAmountInteger = mkrAmount.shiftedBy(MKR_NUMBER_OF_DIGITS).toFixed(0);
+    await govContract['approve(address,uint256)'](chiefContract.address, mkrAmountInteger);
+    await chiefContract.lock(mkrAmountInteger);
+    await chiefContract.etch([spellAddress]);
+    const slate = ethers.utils.keccak256(ethers.utils.hexZeroPad(spellAddress, 32));
```

Comment:
should we make it a helper?

Reply:
You mean the encoding part? I wouldn't know what is does here to come up with a proper name. Since I also don't know when and if it will be useful I will skip it.